### DISCUSSION TITLE
[Suggestion] Logic overrides manually toggleable mechanisms

### DISCUSSION THREAD

**Discussion by MadTheDog** [tsunamayo/Starship-EVO#4334] [Suggestions]
The title might be weird, but hear me out: Currently, there are 3 mechanisms that are able to be toggled manually: the Blast door, the trap door, and rails. 
What I am suggesting is to have the option that logic will always override manual input. For example, if you turn a blast door off with a switch it would "lock" it, making it unable to be used by hand. This would make us able to make locks for the blast doors.
I hope I come through clearly, but if you don't understand I'll elaborate more

**Comment by CurioInventorium** [comment]
You could place an AND logic block in-between the activation logic and the rail, piston, etc...
The AND block would have two inputs; One from the from the control logic and the other from a switch that can act as a lock.
Only when the 'lock' switch is on will the logic get passed through.

Hopefully that makes sense. I've only just finished work and my brain is a bit frazzled.

**Comment by MadTheDog** [comment]
Oh yeah I know that, by "manually" I meant by hitting f directly on the mechanism, like the blast door, you can connect it to a switch, but it does also recognise your f- interaction, my suggestion is to be able to disable the ability to use f as an activation method

**Comment by MatCat** [comment]
I am not a fan of this idea, it would break current properly designed systems that properly handle fixed inputs like switches, for example in my circuits I have a special circuit I use that can check switches and flip them to match the state of the circuit if they don't match, to allow logic to override would break complex logic systems for the sake of simplicity, which would water the logic system down.

**Comment by CurioInventorium** [comment]
I guess an alternative would to have a lockable switch. Or a lock/authenticator logic block. The same way doors in Rust can have a lock attached to them.
Maybe even a keypad logic block.

**Comment by MatCat** [comment]
What I do when I want a switch to activate a circuit but NOT continusally give it a signal is to use a pulse generator on it, a simple clock, nor gate, and AND gate, you tie the switch to the timer, and the AND, the timer to the not, and the not to the AND, set the timer to how long you want your generated pulse to be, and the output of the AND gate will be the output of the circuit, so flip the switch ON and a small pulse configured by the timer will fire, rather then a static ON signal, this allows you to have say an OR gate input to the circuit you are controlling, where the switch can send it a pulse to go on, but then so can any other thing without interference from the switch.  Of course if the circuit is externally triggered on or off that switch is going to be decoupled from what the real state is, which could be overcome with a little extra logic.

**Comment by MatCat** [comment]
sorry I said nor then not, a nor gate with a single input is effectively a not gate, so I tend to just call them that when used as such :).

**Comment by tsunamayo** [comment]
It make a lot of sense to prevent manual activation if there is a button linked. There is no implicit OR gate in the game, it is the same principle. I dont see how it could break something, it only concern manual activation @MatCat ... Cheers

**Comment by MadTheDog** [comment]
@tsunamayo I like the idea of having the option to disable it, the idea of having a separate logic block, or just an option in a menu like we can with piston and rotor speed. The option to lock it could be part of the new logic/programming block you have mentioned elsewhere

**Comment by Xerma** [comment]
You'd most likely need to have it be a GUI when interacting with the doors/etc. Due to it needing to be changed ONLY by the one who placed the door - or later down the line with faction permissions. So then you would open up the door GUI and switch it to be logic mandatory or logic optional
